### PR TITLE
Add lint to template

### PR DESCRIPTION
**Pull request by saargt**

### CODE REVIEW

File: templates/server-side/typescript/api.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// The comment below disables the eslint camelcase rule for the whole file.
```
Comment (Chasky-H) [inline templates/server-side/typescript/api.ts:1]: Can you apply a rule forcing all functions to be snake_case for this file?

Comment (saargt) [reply templates/server-side/typescript/api.ts:1]: I can, but I don't think it's a good idea.
In core reaources I have a few adapter functions and these are not exported, and are not snake cased.

What do you @ido-pepperi @ShirMarciano think?

Comment (ShirMarciano) [reply templates/server-side/typescript/api.ts:1]: @saargt  I think he asked it only for the api.ts file
Can we have rule on specific file?

Comment (ShirMarciano) [reply templates/server-side/typescript/api.ts:1]: @saargt  put this rule here in the template and not in the lint

**Comment by saargt** [comment]
@Chasky-H I moved the implementation of lint into cpi and server sides, but kept the root script, so it could be used in the build process easily.

**Comment by saargt** [comment]
1. Don't add lint to entire folder's build - only for each folder.
2. Add package.json on cpi and server-side the npm run lint prior to run build.
3. add fix lint.
4. add snake case in api.ts.
5. Run lint on these files, so that they would be able to build once used.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    "plugins": [
+    ],
+    "rules": {
+        // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+        // e.g. "@typescript-eslint/explicit-function-return-type": "off",
+    },
+    "overrides": [
+        {
+          "files": ["templates/server-side/installation.ts", "templates/server-side/typescript/api.ts"],
```
Comment (ShirMarciano) [inline .eslintrc.json:22]: in the files tree it is look like installation.ts is under typescript folder also

File: index.js [hunk 3]
```diff
@@ -126,31 +126,42 @@ async function chooseAddonMetadata() {
...
+
             const clientManager = clientVersion === '11' ? 'yarn' : 'npm install --force';
+
             let initCommand = useClient ? `cd ./client-side && ${clientManager} && cd .. ` : '';
             initCommand += useCpi ? '&& cd ./cpi-side && npm install --force && cd .. ' : '';
             initCommand += useServer ? '&& cd ./server-side && npm install --force && cd ..' : '';
-            config.scripts.init = initCommand.startsWith('&&') ?  initCommand.slice(2, initCommand.length) : initCommand;
-            config.scripts.build = buildCommand.startsWith('&&') ?  buildCommand.slice(2, buildCommand.length) : buildCommand;
+
+            config.scripts.build = buildCommand;
```
Comment (Chasky-H) [inline index.js:149]: missing `buildCommand.startsWith('&&') ? buildCommand.slice(2, buildCommand.length) : buildCommand`

**Comment by Chasky-H** [comment]
This is great! Nice work @saargt 💪

**Review by ShirMarciano — APPROVED** [review]

**Pull request merged**